Documentation/ApiOverview/Database/ConnectionPool/Index.rst

```diff
@@ -17,23 +18,29 @@ The :php:`QueryBuilder` is the default object used by extension
 authors to express complex queries, while a :php:`Connection` instance can be
 used as a shortcut to handle some simple query cases.
 
-Pooling
-=======
+..  _database-connection-pool-pooling:
+
+Pooling: multiple connections to different database endpoints
+=============================================================
 
 TYPO3 can handle multiple connections to different database endpoints at the
 same time. This can be configured for each individual table in
 :php:`$GLOBALS['TYPO3_CONF_VARS']` (see :ref:`database configuration
 <database-configuration>` for details). This makes it possible to run tables on
 different databases without an extension developer having to worry about it.
 
-The :php:`ConnectionPool` implements this feature: It looks for configured
+The :php-short:`\TYPO3\CMS\Core\Database\ConnectionPool` 
+implements this feature: It looks for configured
 table-to-database mapping and can return a :php:`Connection` or a
-:php:`QueryBuilder` instance for that specific connection. These objects know
+:php-short:`\TYPO3\CMS\Core\Database\Query\QueryBuilder` instance 
+for that specific connection. These objects know
 internally which target connection they are dealing with and will quote field
 names accordingly, for instance.
 
+..  _database-connection-pool-beware:
+
 Beware
-------
+======
```

Comment:
The beware is related to the pooling, so it is a subsection of it. Maybe we should convert this to a note or attention?

Reply:
I considered it but there is so much text and already another info box. Reverting my change here for now